### PR TITLE
[idrac_secure_boot]: Attributes, Reset Keys and Export secure boot certificates

### PR DESCRIPTION
**Pull request by ABHISHEK-SINHA10**
# Description
A few sentences describing the overall goals of the pull request's commits.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# ISSUE TYPE

- Bugfix Pull Request
- Docs Pull Request
- Feature Pull Request
- Test Pull Request

##### OUTPUT

```paste below

```
##### ADDITIONAL INFORMATION

```paste below

```
# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

### CODE REVIEW

**Comment by ansible-collections-svc** [comment]
Can one of the admins verify this patch?

**Comment by sachin-apa** [comment]
run e2e

File: docs/modules/idrac_secure_boot.rst [hunk 1]
```diff
@@ -135,7 +229,10 @@ Notes
 
 .. note::
    - This module will always report changes found to be applied when run in \ :literal:`check mode`\ .
-   - This module does not support idempotency when \ :emphasis:`import\_certificates`\  is provided.
+   - This module does not support idempotency when \ :emphasis:`reset\_type`\  or \ :emphasis:`export\_certificates`\  or \ :emphasis:`import\_certificates`\  is provided.
+   - The order of operations is as follows configure Secure Boot settings(boot\_mode, secure\_boot, secure\_boot\_mode, secure\_boot\_policy, force\_int\_10), export, certificate reset, import, followed by idrac reset.
+   - \ :emphasis:`export\_certificate`\  will export all the certificates of the key defined in the playbook.
+   - This module considers values of \ :emphasis:`restart`\ , \ :emphasis:`job\_wait`\  only for the last operation in the sequence.
```
Comment (reetivel) [inline docs/modules/idrac_secure_boot.rst]: This module considers values of I(restart) and  I(job_wait) only for the last operation in the sequence.

File: docs/modules/idrac_secure_boot.rst [hunk 2]
```diff
@@ -36,17 +102,43 @@ Parameters
   platform_key (optional, path, None)
     The absolute path of the Platform key certificate file for UEFI secure boot.
 
+    Directory path with write permissions if \ :emphasis:`export\_certificates`\  is \ :literal:`true`\ .
```
Comment (reetivel) [inline docs/modules/idrac_secure_boot.rst]: Directory path with write permission when I(export_certificates) is C(true).

File: docs/modules/idrac_secure_boot.rst [hunk 3]
```diff
@@ -36,17 +102,43 @@ Parameters
...
+
 
   KEK (optional, list, None)
     A list of absolute paths of the Key Exchange Key (KEK) certificate file for UEFI secure boot.
 
+    Directory path with write permissions if \ :emphasis:`export\_certificates`\  is \ :literal:`true`\ .
```
Comment (reetivel) [inline docs/modules/idrac_secure_boot.rst]: Directory path with write permission when I(export_certificates) is C(true).

File: docs/modules/idrac_secure_boot.rst [hunk 4]
```diff
@@ -36,17 +102,43 @@ Parameters
...
+
 
   database (optional, list, None)
-    A list of absolute paths of the Database certificate file for UEFI secure boot.
+    A list of absolute paths of the Allowe Database(DB) certificate file for UEFI secure boot.
+
+    Directory path with write permissions if \ :emphasis:`export\_certificates`\  is \ :literal:`true`\ .
```
Comment (reetivel) [inline docs/modules/idrac_secure_boot.rst]: Directory path with write permission when I(export_certificates) is C(true).

File: docs/modules/idrac_secure_boot.rst [hunk 5]
```diff
@@ -36,17 +102,43 @@ Parameters
...
 
 
   disallow_database (optional, list, None)
-    A list of absolute paths of the Disallow Database certificate file for UEFI secure boot.
+    A list of absolute paths of the Disallow Database(DBX) certificate file for UEFI secure boot.
+
+    Directory path with write permissions if \ :emphasis:`export\_certificates`\  is \ :literal:`true`\ .
```
Comment (reetivel) [inline docs/modules/idrac_secure_boot.rst]: Directory path with write permission when I(export_certificates) is C(true).

File: docs/modules/idrac_secure_boot.rst [hunk 6]
```diff
@@ -27,6 +37,62 @@ The below requirements are needed on the host that executes this module.
...
+
+
+  secure_boot_policy (optional, str, None)
+    The following are the types of Secure Boot policy.
+
+    \ :literal:`Custom`\  inherits the standard certificates and image digests that are loaded in the system by default. You can modify the certificates and image digests.
+
+    \ :literal:`Standard`\  indicates that the system has default certificates, image digests, or hash loaded from the factory.
+
+    When Secure Boot Policy is configured as Custom you can perform the following operations as view, export, import, delete, delete all, and reset policies.
```
Comment (reetivel) [inline docs/modules/idrac_secure_boot.rst]: When Secure Boot Policy is configured as Custom, you can perform the following operations: view, export, import, delete, delete all, and reset policies.

File: docs/modules/idrac_secure_boot.rst [hunk 7]
```diff
@@ -27,6 +37,62 @@ The below requirements are needed on the host that executes this module.
...
+
+
+  force_int_10 (optional, str, None)
+    Determines if the system BIOS will load the legacy video (INT 10h) option ROM from the video controller.
```
Comment (reetivel) [inline docs/modules/idrac_secure_boot.rst]: Determines whether the system BIOS loads the legacy video (INT 10h) option ROM from the video controller.

File: docs/modules/idrac_secure_boot.rst [hunk 8]
```diff
@@ -36,17 +102,43 @@ Parameters
...
 
   KEK (optional, list, None)
     A list of absolute paths of the Key Exchange Key (KEK) certificate file for UEFI secure boot.
 
+    Directory path with write permissions if \ :emphasis:`export\_certificates`\  is \ :literal:`true`\ .
+
 
   database (optional, list, None)
-    A list of absolute paths of the Database certificate file for UEFI secure boot.
+    A list of absolute paths of the Allowe Database(DB) certificate file for UEFI secure boot.
```
Comment (reetivel) [inline docs/modules/idrac_secure_boot.rst]: "Allow"

File: docs/modules/idrac_secure_boot.rst [hunk 9]
```diff
@@ -135,7 +229,10 @@ Notes
 
 .. note::
    - This module will always report changes found to be applied when run in \ :literal:`check mode`\ .
```
Comment (reetivel) [inline docs/modules/idrac_secure_boot.rst]: When this module runs in check_mode for I(import_certificates), it reports the list of changes found.

File: playbooks/idrac/idrac_secure_boot.yml [hunk 10]
```diff
@@ -4,6 +4,53 @@
...
+    - name: Reset Secure Boot certificates.
+      dellemc.openmanage.idrac_secure_boot:
+        idrac_ip: "192.168.1.2"
+        idrac_user: "user"
+        idrac_password: "password"
+        ca_path: "/path/to/ca_cert.pem"
+        reset_keys: "ResetAllKeysToDefault"
+      delegate_to: localhost
+
+    - name: Export multiple SecureBoot certificate.
```
Comment (reetivel) [inline playbooks/idrac/idrac_secure_boot.yml]: - name: Export multiple Secure Boot certificates.
  
  (Please maintain consistency with the use of  " Secure Boot". )

File: playbooks/idrac/idrac_secure_boot.yml [hunk 11]
```diff
@@ -4,6 +4,53 @@
...
+        platform_key: /user/name/export_cert/pk
+        KEK:
+          - /user/name/export_cert/kek
+        database:
+          - /user/name/export_cert/db
+        disallow_database:
+          - /user/name/export_cert/dbx
+      delegate_to: localhost
+
     - name: Import multiple SecureBoot certificate without applying to iDRAC.
```
Comment (reetivel) [inline playbooks/idrac/idrac_secure_boot.yml]: - name: Import multiple Secure Boot certificates without applying to iDRAC.
  
  (Please maintain consistency with the use of  " Secure Boot". )

File: docs/modules/idrac_secure_boot.rst [hunk 12]
```diff
@@ -148,6 +245,49 @@ Examples
...
+
+    - name: Reset Secure Boot certificates.
+      dellemc.openmanage.idrac_secure_boot:
+        idrac_ip: "192.168.1.2"
+        idrac_user: "user"
+        idrac_password: "password"
+        ca_path: "/path/to/ca_cert.pem"
+        reset_keys: "ResetAllKeysToDefault"
+
+    - name: Export multiple SecureBoot certificate.
```
Comment (reetivel) [inline docs/modules/idrac_secure_boot.rst]: - name: Export multiple Secure Boot certificates.

File: docs/modules/idrac_secure_boot.rst [hunk 13]
```diff
@@ -148,6 +245,49 @@ Examples
...
+        export_certificates: true
+        platform_key: /user/name/export_cert/pk
+        KEK:
+          - /user/name/export_cert/kek
+        database:
+          - /user/name/export_cert/db
+        disallow_database:
+          - /user/name/export_cert/dbx
+
     - name: Import multiple SecureBoot certificate without applying to iDRAC.
```
Comment (reetivel) [inline docs/modules/idrac_secure_boot.rst]: - name: Import multiple Secure Boot certificates without applying to iDRAC.

**Review by Saksham-Nautiyal — APPROVED** [review]
Validation completed for export.

**Comment by Saksham-Nautiyal** [comment]
Validation completed for Attributes

**Review by reetivel — COMMENTED** [review]
Added review comments

File: playbooks/idrac/idrac_secure_boot.yml (hunk 11 above)
Comment (ABHISHEK-SINHA10) [reply playbooks/idrac/idrac_secure_boot.yml]: Done

File: playbooks/idrac/idrac_secure_boot.yml (hunk 10 above)
Comment (ABHISHEK-SINHA10) [reply playbooks/idrac/idrac_secure_boot.yml]: Done

File: docs/modules/idrac_secure_boot.rst (hunk 9 above)
Comment (ABHISHEK-SINHA10) [reply docs/modules/idrac_secure_boot.rst]: Corrected to this -->  When this module runs in check_mode for I(import_certificates), it reports changes found.

Comment (singh-lovepreet1) [reply docs/modules/idrac_secure_boot.rst]: This message is kept as "This module will always report changes found to be applied for I(import_certificates) when run in C(check mode)."

File: docs/modules/idrac_secure_boot.rst [hunk 14]
```diff
@@ -135,7 +229,10 @@ Notes
 
 .. note::
    - This module will always report changes found to be applied when run in \ :literal:`check mode`\ .
-   - This module does not support idempotency when \ :emphasis:`import\_certificates`\  is provided.
+   - This module does not support idempotency when \ :emphasis:`reset\_type`\  or \ :emphasis:`export\_certificates`\  or \ :emphasis:`import\_certificates`\  is provided.
+   - The order of operations is as follows configure Secure Boot settings(boot\_mode, secure\_boot, secure\_boot\_mode, secure\_boot\_policy, force\_int\_10), export, certificate reset, import, followed by idrac reset.
```
Comment (reetivel) [inline docs/modules/idrac_secure_boot.rst]: To configure the secure boot settings, the idrac_secure_boot module performs the following order of operations— set attributes, export certificate, reset keys, import certificate, and restart iDRAC.

File: docs/modules/idrac_secure_boot.rst (hunk 1 above)
Comment (singh-lovepreet1) [reply docs/modules/idrac_secure_boot.rst]: Updated

File: docs/modules/idrac_secure_boot.rst (hunk 3 above)
Comment (singh-lovepreet1) [reply docs/modules/idrac_secure_boot.rst]: Updated

File: docs/modules/idrac_secure_boot.rst (hunk 5 above)
Comment (singh-lovepreet1) [reply docs/modules/idrac_secure_boot.rst]: Updated

File: docs/modules/idrac_secure_boot.rst (hunk 7 above)
Comment (singh-lovepreet1) [reply docs/modules/idrac_secure_boot.rst]: Updated

File: docs/modules/idrac_secure_boot.rst (hunk 12 above)
Comment (singh-lovepreet1) [reply docs/modules/idrac_secure_boot.rst]: Updated

File: docs/modules/idrac_secure_boot.rst (hunk 13 above)
Comment (singh-lovepreet1) [reply docs/modules/idrac_secure_boot.rst]: updated

File: docs/modules/idrac_secure_boot.rst (hunk 14 above)
Comment (singh-lovepreet1) [reply docs/modules/idrac_secure_boot.rst]: Updated

File: docs/modules/idrac_secure_boot.rst (hunk 8 above)
Comment (singh-lovepreet1) [reply docs/modules/idrac_secure_boot.rst]: Done

File: docs/modules/idrac_secure_boot.rst (hunk 6 above)
Comment (singh-lovepreet1) [reply docs/modules/idrac_secure_boot.rst]: Updated

File: docs/modules/idrac_secure_boot.rst (hunk 4 above)
Comment (singh-lovepreet1) [reply docs/modules/idrac_secure_boot.rst]: Updated

File: docs/modules/idrac_secure_boot.rst (hunk 2 above)
Comment (singh-lovepreet1) [reply docs/modules/idrac_secure_boot.rst]: Updated

**Review by rajshekarp87 — DISMISSED** [review]

**Review by rajshekarp87 — APPROVED** [review]

**Review by Saksham-Nautiyal — APPROVED** [review]

**Review by ShivamSh3 — APPROVED** [review]

**Pull request merged**